packages/integration-karma/test/light-dom/events/index.spec.js

```diff
@@ -0,0 +1,208 @@
+import { createElement, setFeatureFlagForTest } from 'lwc';
+import { extractDataIds } from 'test-utils';
+
+import LightChild from 'x/lightChild';
+import ShadowContainer from 'x/shadowContainer';
+import LightContainer from 'x/lightContainer';
+
+function createTestElement(tag, component) {
+    const elm = createElement(tag, { is: component });
+    elm.setAttribute('data-id', tag);
+    document.body.appendChild(elm);
+    return extractDataIds(elm);
+}
+
+function dispatchEventWithLog(target, nodes, event) {
+    const log = [];
+
+    [...Object.values(nodes), document.body, document.documentElement, document, window].forEach(
+        (node) => {
+            node.addEventListener(event.type, (event) => {
+                log.push([event.currentTarget, event.target, event.composedPath()]);
+            });
+        }
+    );
```

Comment:
This leaks event listeners even after the test has run. We should add an `afterEach` to clean these up.

Reply:
Does it matter since we clean the DOM after every test?

Reply:
No, because the DOM is not cleaned in that way. The cleanup code checks for any added DOM nodes in the head/body and removes them afterward, but it doesn't check for event listeners. (In fact I don't think it can, unless it overrides `Element.prototype.addEventListener` in the `beforeEach`.)